sdk/eventhub/event-hubs/src/eventhubConnectionConfig.ts

```diff
@@ -141,6 +145,22 @@ export const EventHubConnectionConfig = {
     return config as EventHubConnectionConfig;
   },
 
+  /**
+   * Updates the provided EventHubConnectionConfig to use the custom endpoint address.
+   * @param config An existing connection configuration to be updated.
+   * @param customEndpointAddress The custom endpoint address to use.
+   */
+  setCustomEndpointAddress(config: EventHubConnectionConfig, customEndpointAddress: string): void {
+    // The hostname should match the host prior to using the custom endpoint.
+    config.hostname = config.host;
+    const { hostname, port } = new URL(customEndpointAddress);
```

Comment:
Is it worth checking that the string is not empty? Or that hostname doesn't come back null, for instance?

Reply:
We do check that the string is not empty before calling this method. If the customEndpointAddress isn't a valid then URL will throw a TypeError indicating the url is invalid (so I don't think I can come up with a better error :smile:)